brig/cmd/brig/commands/dashboard.go

```diff
@@ -0,0 +1,99 @@
+package commands
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/Azure/brigade/pkg/portforwarder"
+	"github.com/bacongobbler/browser"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	dashboardUsage = `Opens the Kashti dashboard with the default browser.
+`
+	remoteAPIPort = 7745
+	remotePort    = 80
+)
+
+var (
+	port            int
+	apiPort         int
+	kashtiNamespace string
+	openDashboard   bool
+)
+
+func init() {
+	Root.AddCommand(dashboard)
+
+	flags := dashboard.PersistentFlags()
+	flags.IntVar(&port, "port", 8081, "local port for the Kashti dashboard")
+	flags.IntVar(&apiPort, "api-port", 7745, "local port for the Brigade API server")
+	flags.StringVar(&kashtiNamespace, "kashti-namespace", "default", "namespace for Kashti")
+	flags.BoolVar(&openDashboard, "open-dashboard", true, "open the dashboard in the browser")
+}
+
+var dashboard = &cobra.Command{
+	Use:   "dashboard",
+	Short: "Open the Kashti dashboard",
+	Long:  dashboardUsage,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		log.Printf("Connecting to kashti at http://localhost:%d...\n", port)
+
+		tunnel, err := startProxy(port)
+		if err != nil {
+			return err
+		}
+		defer tunnel.Close()
+
+		stop := make(chan os.Signal, 2)
+		signal.Notify(stop, os.Interrupt, syscall.SIGTERM)
+		go func() {
+			<-stop
+			os.Exit(0)
+		}()
+
+		log.Println("Connected! When you are finished with this session, enter CTRL+C.")
+
+		if openDashboard {
+			if err := browser.Open(fmt.Sprintf("http://localhost:%d", port)); err != nil {
+				return err
+			}
+		}
+
+		// block until the user sends a CTRL+C
+		for {
```

Comment:
The empty `for` loop here always blocks - at least when compiled with my Go versions (`go version go1.9.2 windows/amd64`, `go version go1.10.3 linux/amd64` and I think on `darwin` as well) - meaning it opens the browser, but the proxy doesn't correctly starts.

Adding instructions here (printing, waiting) seems to avoid the issue - will come back after investigating a bit.